R/hand_tiles.R

```diff
@@ -0,0 +1,413 @@
+    tiles      <- get_hand_xy(bbx)
+
+    hand_urls  <- sprintf("%s/hand/0/%s/%s.tif",
+                          base_url,
+                          tiles[, 2],
+                          tiles[, 1])
+
+    hand_list  <- vector("list", length = nrow(tiles))
+
+    pb  <- progress::progress_bar$new(
+        format = "Downloading HAND [:bar] :percent eta: :eta",
+        total  = length(hand_urls),
+        clear  = FALSE,
+        width  = 60
+    )
+
+    for (i in seq_along(hand_urls)) {
+        pb$tick()
+        tmpfile <- tempfile(fileext = ".tif")
+        resp <- httr::GET(
+            hand_urls[i],
+            httr::user_agent(
+                "FloodMapping (https://github.com/mikejohnson51/FloodMapping)"
+            ),
+            httr::write_disk(tmpfile, overwrite = TRUE),
+            ...
+        )
+
+        if (!grepl("image/tif", httr::http_type(resp))) {
+            stop(
+                paste("This url:", hand_urls[i], "did not return a tif"),
+                .call = FALSE
+            )
+        }
+
+        hand_list[[i]] <- tmpfile
+    }
+
+    merge_hand_tiles(hand_list, target_prj = prj)
+}
+
+#' @title Merge Rasters
+#'
+#' @description
+#' Merge multiple downloaded raster files into a single file.
+#' The input `target_prj` describes the projection for the new grid.
+#'
+#' **Note: this function is for internal use. Use \link{get_hand_raster}.**
+#'
+#' @source Attribution: The `elevatr` R package.
+#'         \url{https://github.com/jhollist/elevatr}
+#'
+#' @param raster_list a list of raster file paths to be mosaiced
+#' @param target_prj the target projection of the output raster
+#' @param method the method for resampling/reprojecting. Default is 'bilinear'.
+#'               Options can be found on the GDAL documentation site
+#'               for `gdalwarp`.
+#' @param returnRaster if TRUE, return a raster object (default),
+#'                     else, return the file path to the object
+#' @export
+#' @keywords internal
+merge_hand_tiles <- function(raster_list,
+                             target_prj,
+                             method = "bilinear",
+                             return_raster = TRUE) {
+
+    message("Mosaicing & Projecting")
+
+    destfile <- tempfile(fileext = ".tif")
+    files    <- unlist(raster_list)
+
+    if (is.null(target_prj)) {
+        r <- raster::raster(files[1])
+        target_prj <- raster::crs(r)
+    }
+
+    sf::gdal_utils(
+        util = "warp",
+        source = files,
+        destination = destfile,
+        options = c(
+            "-t_srs",
+            as.character(target_prj),
+            "-r",
+            method
+        )
+    )
+
+    if (return_raster) {
+        raster::raster(destfile)
+    } else {
+        destfile
+    }
+}
+
+#' @title Get HAND Tile XY Coordinates
+#' @param aoi Area of interest
+#' @keywords internal
+get_hand_xy <- function(aoi) {
+    # Extent of HAND/Catchmask Rasters
+    ext <- data.frame(xmin = -89.95326,
```

Comment:
Is there a way to not have this hard coded? Or to warn if a user requests outside this bounds?

Reply:
I can't think of a way to have the extent dynamically acquired without having access to the seamless raster layer. If you have any ideas, I can definitely implement them :smile:

As a compromise, I will issue a warning if the bounding box for an AOI falls outside of the extent!